1_network/src/reach_classification.R

```diff
@@ -0,0 +1,136 @@
+    filter(!to_seg %in% unique(segs_single$subseg_seg))
+
+  # calculate combined distances
+
+  downstream1 <- downstream %>% pull(to_seg)
+
+  # downstream1 distance
+  # dist <- select(downstream, seg1 = subseg_seg, seg2 = to_seg, dist_down_reservoir, GRAND_ID) %>%
+  #   left_join(st_drop_geometry(select(network$edges, seg2 = subseg_seg, subseg_length))) %>%
+  #   rowwise() %>%
+  #   mutate(dist_down_reservoir2 = sum(dist_down_reservoir, subseg_length)) %>%
+  #   select(-subseg_length)
+
+  downstream2 <- filter(res, subseg_seg %in% downstream1) %>%
+    filter(!to_seg %in% segs_single$subseg_seg) %>%
+    pull(to_seg)
+
+  # dist2
+  # dist_2 <- filter(res, subseg_seg %in% downstream1) %>%
+  #   filter(!to_seg %in% segs_single$subseg_seg) %>%
+  #   select(seg2 = subseg_seg, seg3 = to_seg) %>%
+  #   left_join(select(res, seg3 = subseg_seg, subseg_length)) %>%
+  #   right_join(dist) %>%
+  #   rowwise() %>%
+  #   mutate(dist_down_reservoir3 = sum(dist_down_reservoir2, subseg_length)) %>%
+  #   select(-subseg_length)
+
+  downstream12_reservoirs <- downstream %>% pull(GRAND_ID)
+
+  down1_cross <- data.frame(subseg_seg = as.character(downstream1), closest_GRAND_ID = downstream12_reservoirs) %>%
+    filter(!is.na(subseg_seg))
+  down2_cross <- data.frame(subseg_seg = as.character(downstream2), closest_GRAND_ID = downstream12_reservoirs) %>%
+    filter(!is.na(subseg_seg))
+
+  # grab upstream reaches
+  upstream1 <- segs_single$from_segs[segs_single$type %in% c('contains_reservoir', 'reservoir_inlet_reach')]
+
+  upstream1_reservoirs <- segs_single$GRAND_ID[segs_single$type %in% c('contains_reservoir', 'reservoir_inlet_reach')]
+
+  up1_cross <- data.frame(closest_GRAND_ID = rep(upstream1_reservoirs, grepl(';', upstream1) + 1),
+                          subseg_seg = unlist(stringr::str_split(upstream1, ';'))) %>%
+    filter(!subseg_seg %in% '') %>%
+    filter(!subseg_seg %in% segs_single$subseg_seg) %>%
+    distinct()
+  # seg 262 is upstream of two different reservoirs. Which should it be associated with?
+
+  upstream2 <- res$from_segs[res$subseg_seg %in% up1_cross$subseg_seg]
+  up2_cross <- data.frame(closest_GRAND_ID = rep(up1_cross$closest_GRAND_ID, grepl(';', upstream2) + 1),
+                          subseg_seg = unlist(stringr::str_split(upstream2, ';'))) %>%
+    filter(!subseg_seg %in% '') %>%
+    filter(!subseg_seg %in% segs_single$subseg_seg) %>%
+    distinct()
+
+  res_cross <- bind_rows(down1_cross, down2_cross, up1_cross, up2_cross)
+
+  # create distance upstream metric
+  # from_to <- select(segs_single, subseg_seg, from_segs, dist_down_reservoir, GRAND_ID) %>%
+  #   tidyr::separate_rows('from_segs', sep = ';') %>%
+  #   rename(seg_up1 = from_segs) %>%
+  #   filter(!seg_up1 %in% segs_single$subseg_seg)
+
+  # now use above info to classify reaches
+  # data frame
+  reach_class = select(network$edges, seg_id_nat, subseg_seg) %>%
```

Comment:
Any reason not to use `<-` here, just for consistency?

Reply:
Yikes! Haven't done that in a while!